celery_serverless/invoker.py

```diff
@@ -219,3 +238,57 @@ def invoke(target='watchdog', config=None, *args, **kwargs):
 
 def invoke_worker(config=None, data=None, *args, **kwargs):
     return invoke(target='worker', extra_data=data or {}, *args, **kwargs)
+
+
+def invoke_watchdog(config=None, data=None, check_lock=True, *args, **kwargs):
+    from .watchdog import _get_watchdog_lock
+
+    if check_lock:
+        lock_watchdog = _get_watchdog_lock(enforce=False)[0]
+        if lock_watchdog.locked():
+            logger.info('Watchdog lock already held. Giving up.')
+            return False, RuntimeError('Watchdog lock already held')
+    return invoke(target='watchdog', extra_data=data or {}, *args, **kwargs)
+
+
+def client_invoke_watchdog(check_lock=True, blocking_lock=False, *args, **kwargs):
+    if not check_lock:
+        logger.debug('Not checking client lock')
+        return invoke_watchdog(check_lock=True, *args, **kwargs)
+
+    client_lock = _get_client_lock()[0]
+    locked = client_lock.acquire(blocking_lock)
+    if not locked:
+        logger.info('Could not get Client lock. Giving up.')
+        return False, RuntimeError('Client lock already held')
+
+    logger.debug('Got the client lock')
+    try:
+        return invoke_watchdog(with_lock=True, *args, **kwargs)
+    finally:
+        logger.debug('Releasing the client lock')
+        client_lock.release()
+
+
+def _get_client_lock(lock_url='', lock_name='', default=multiprocessing.Lock, enforce=False):
+    if _CLIENT_LOCK and lock_name == _CLIENT_LOCK['lock_name']:
+        return _CLIENT_LOCK['lock'], _CLIENT_LOCK['lock_name']
+
+    lock_name = lock_name or os.environ.get('CELERY_SERVERLESS_CLIENT_LOCK_NAME', 'celery_serverless:watchdog-client')
+    lock_url = lock_url or os.environ.get('CELERY_SERVERLESS_LOCK_URL', '(unavailable)')
+    if enforce:
+        if lock_url == 'disabled':
+            lock_url = ''
+        else:
+            assert lock_url, 'The CELERY_SERVERLESS_LOCK_URL envvar should be set. Even to "disabled" to disable it.'
+
+    if default and not lock_url:
```

Comment:
`default_lock`? Maybe this variable should always be a string and you could check for `multiprocessing`, since the redis version is a remote url.

Reply:
This born as a copy from `_get_watchdog_lock`.

Generalizing both will make the `default` more reasonable...

However, I had not understood your "be a string and check for multiprocessing". You mean receiving `lock_url='multiprocessing'`?

Reply:
Yep, I think it is strange that sometimes it is a class, sometimes a url.

Reply:
@sbneto Take a look on how `_get_lock` is now 👍